src/main/kotlin/pt/up/fe/ni/website/backend/service/activity/AbstractActivityService.kt

```diff
@@ -74,13 +79,40 @@ abstract class AbstractActivityService<T : Activity>(
     fun removeTeamMemberById(idActivity: Long, idAccount: Long): T {
         val activity = getActivityById(idActivity)
         if (!accountService.doesAccountExist(idAccount)) {
-            throw NoSuchElementException(
-                ErrorMessages.accountNotFound(
-                    idAccount
-                )
-            )
+            throw NoSuchElementException(ErrorMessages.accountNotFound(idAccount))
         }
         activity.teamMembers.removeIf { it.id == idAccount }
         return repository.save(activity)
     }
+
+    fun addGalleryImage(activityId: Long, image: MultipartFile): Activity {
+        val activity = getActivityById(activityId)
+
+        var imageFolder = "activities"
+
+        if (activity is Event) {
+            imageFolder = EventService.IMAGE_FOLDER
+        } else if (activity is Project) {
+            imageFolder = ProjectService.IMAGE_FOLDER
+        }
```

Comment:
I think this is a big violation of OOP and we can improve this. My suggestion is to a method like this in the abstract activity service:

```suggestion
fun addGalleryImage(activityId: Long, image: MultipartFile, imageFolder: String): Activity {
        val activity = getActivityById(activityId)
        
        val fileName = fileUploader.buildFileName(image, activity.title)
        val imageName = fileUploader.uploadImage(imageFolder + "/gallery", fileName, image.bytes)

        activity.gallery.add(imageName)

        return repository.save(activity)
    }
```

And then add new methods to the project/events services where the name is chosen. This would maintain the logic and decouple the inheritance logic from the abstract service

Reply:
Changed the highlighted code as proposed, while keeping the companion objects.